### ISSUE TITLE
improving drumkits

### ISSUE THREAD

**Issue by trebmuh** [hydrogen-music/hydrogen#300]
It would be awesome to make the drumkit definition to be translatable since this is a part of the GUI.

**Comment by thijz** [comment]
this will be hard to do since the drumkits are not part of the code + anyone can set up a drumkit repo

**Comment by trebmuh** [comment]
I've got 2 arguments:
* first: some of the drumkit are hold by hydrogen (sourceforge etc...) so it could be done at least for those
* second: I was more thinking about "offering the possibility" of the drumkit information to be translated rather than "doing that inside Hydrogen". As far as I know, there is no field in the drumkit.xml for translation. Then putting a field like that would leave the possibility for the drumkit to be translated. This would then be up to the drumkit maker and to a translator to talk together. This way, it would become "doable" (as opposed to now where the situation doesn't allow translation to be made even if the translator and the drumkit maker are already in touch). Then, a part of the code which could get the language environment and then check if a translation is done in the drumkit.xml in this language - if not, it could default to the "original" language of the drumkit description field - would do the tricks.
I think you closed this thread a bit too fast without any discussion :)

**Comment by mikotoiii** [comment]
I might be misunderstanding this, but it sounds to me like the translations
should be in an xml file, and then someone can set up a repo for the
translated kits. So French kits could be at fr.myhydrogenkits.org while
German kits might be at kits. Ilovehydrogen.de, etc

**Comment by trebmuh** [comment]
That's another idea. A workaround.

Mine was to have a possiblity of several fields in the xml file such as something like that (as an example):
*<info>Based on sounds from famous and classic beatboxes of 1970s and 1980s - Roland TR-606, TR-33 and Boss DR-55.</info>
*<info[fr]>Basé sur les sons des fameuses beatboxes classiques des années 70 et 80 - Roland TR-606, TR-33 et Boss DR-55.</info[fr]>
*<info[de]>...</info[de]>
Then, Hydrogen would check if there is a info field with a language matching the environment language currently used (and defaults to the original one in case of not).

Multiplying copies of drumkits repos just to change a description language - which is more or less 5 lines of descriptions for the drumkit - is a waste of time in my opinion. It sounds better to have the possibility inside the xml file, isn't it ?

Hope to be understandable.
Olivier

**Comment by mauser** [comment]
Yes, this could technically be done.
The xml would need a new tag (which goes parallel to the <info> tag, like <infoTranslatable lang="DE"></infoTranslatable>). It is import to use a new tag, otherwise we would break compatibility with older version of hydrogen which are using that repository.

**Comment by trebmuh** [comment]
Thanks @mauser . I do agree, this is important to keep backward compatibility as long as it can be kept easily.

**Comment by thijz** [comment]
extending the XML sounds ok, but that does make it harder for translators since the text that needs to be translated does not reside in the hydrogen repo (and we want to keep it as easy as possible to translate the application)
any ideas on how to get around this ?

**Comment by mauser** [comment]
The drumkits are not a part of hydrogen and are designed to be self-contained. With this design, even third-party tools can operate on them without having any connection to hydrogen or use stuff from hydrogen's repository. Honestly, i don't want to break that design just make it easier to translate them.

**Comment by trebmuh** [comment]
My point of view (as a translator): it doesn't need to be related to Hydrogen. Just:
1. having the possibility inside the xml to ship translations
2. having Hydrogen looking if a translation in the current environment language exists
would be an awesome starting point.

**Comment by mikotoiii** [comment]
Just throwing this out there: What if there was an optional additional file that Hydrogen could look at for translations in the data/drumkits/xKit directory? something like i18n.xml which had fields that Hydrogen can read and use to apply translations based on locale. That in theory shouldn't interfere with anything as Hydrogen and other programs will still be able to use the drumkit.xml file, and Hydrogen could benefit from having the fields translated/translatable.

It's not really an optimal solution, but it's an idea at least.

**Comment by trebmuh** [comment]
Yes, whether in the xml file or another file in the data/drumkits/xKit (as you proposed above) looks fine to me. I'd leave one or another in coder's hands.

**Comment by trebmuh** [comment]
Could someone with the relevant rights on this repo remove the "wont' fix" tag please ?

**Comment by trebmuh** [comment]
thanks thijz
